src/mm_character.cpp

```diff
@@ -329,26 +331,33 @@ void MMCharacter::_update_query(double delta_t) {
         _force_transition = false;
 
         // Fill query_input with data from the controller
-        MMQueryInput query_input;
-        _fill_query_input(query_input);
+        Ref<MMQueryInput> query_input;
+        query_input.instantiate();
+        _fill_query_input(**query_input);
+
+        if (_animation_tree) {
+            _animation_tree->set(
+                "parameters/" + MMAnimationNode::MOTION_MATCHING_INPUT_PARAM,
+                query_input);
+        }
 
-        // Run query
-        const MMQueryOutput result = query(query_input);
+        // // Run query
```

Comment:
@fire the part that is missing is to bring this code over to `MMAnimationNode`. The thing is, the animation node can't just do `_animation_player->play(...)`, it needs to call `AnimationNode::blend_animation`, much like what `AnimationNodeAnimation` does. This is the API I'm not as familiar with, I'll try to play around with it a bit today.